### PR TITLE
Add versioned link component

### PR DESCRIPTION
**Pull request by jlvandenhout**
# Description of change

This adds a versioned link component. It accepts the same properties as a regular Link component, except that it automatically resolves any path provided by the `to` property to a preferred version when possible.

## Links to any relevant issues

#631 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Changed the iota.rs library link to use this component and switched versions, then checked if the version in the card was updated.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested the wiki locally and tested that all external links work

### CODE REVIEW

File: src/components/VersionedLink.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    const { preferredVersion } = useWikiPreferredVersion(pathname, pluginIds);
+
+    if (preferredVersion !== null) {
+      return <Link to={`${preferredVersion.path}/${docsPath}`} {...props} />
+    }
+  }
+  
+  return <Link to={to} {...props} />
+}
```
Comment (Dr-Electron) [inline src/components/VersionedLink.tsx:22]:
```suggestion
}

```

Triggered 😅

**Comment by Dr-Electron** [comment]
LGTM. Let's try to implement this in the landing page in the next days and test it to see where it makes sense

**Comment by jlvandenhout** [comment]
@Dr-Electron I'm fine merging it now so we can use it in other PRs. We're going to use it for certain anyway.

**Comment by Dr-Electron** [comment]
Ok

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**